projects/leptonica/enhance_fuzzer.cc

```diff
@@ -0,0 +1,57 @@
+/*
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+*/
+
+#include "string.h"
+#include "allheaders.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+
+extern "C" int
+LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+        if(size<3) return 0;
+
+        char filename[256];
+        sprintf(filename, "/tmp/libfuzzer.%d", getppid());
+
+        FILE *fp = fopen(filename, "wb");
+        if (!fp) return 0;
+        fwrite(data, size, 1, fp);
+        fclose(fp);
+
+        PIX *pix, *pix0, *pix1, *pix2, *pix3, *pix4;
+
+        pix = pixRead(filename);
```

Comment:
Using `pixReadMem` could simplify the code and avoid the need for a temporary file.

Reply:
I suggest to check the return value and skip the next lines if it is `NULL`.